packages/expect/src/utils.ts

```diff
@@ -16,3 +21,25 @@ export function recordAsyncExpect(test: any, promise: Promise<any> | PromiseLike
 
   return promise
 }
+
+export function wrapSoft(utils: Chai.ChaiUtils, fn: (this: Chai.AssertionStatic & Assertion, ...args: any[]) => void) {
+  return function (this: Chai.AssertionStatic & Assertion, ...args: any[]) {
+    const test = utils.flag(this, 'vitest-test')
+
+    const state: MatcherState = test?.context._local
+      ? test.context.expect.getState()
+      : getState((globalThis as any)[GLOBAL_EXPECT])
+
+    if (!state.soft)
+      return fn.apply(this, args)
+
+    try {
+      return fn.apply(this, args)
+    }
+    catch (err) {
+      test.result.state = 'fail'
```

Comment:
I am getting this error because test is undefined:

![Screenshot 2023-06-06 at 12 17 25](https://github.com/vitest-dev/vitest/assets/16173870/58f569e2-9c61-4e85-a167-f785b7db9e25)

Reply:
Oh, I got it, need to do something to avoid this

Reply:
Yeah, some kind of a guard with a nice error that `expect.soft` should be called only inside a test